### ISSUE TITLE
Mobs running in place/invisible, thrown/dropped items floating on dedicated server

### ISSUE THREAD

**Issue by gunnerysurgeon** [anegostudios/VintageStory-Issues#4552]
### Game Version

1.20.0-RC1

### Platform

Windows client + Ubuntu Server

### Description

I am running a dedicated server on Linux and 2 of my friends (windows clients) see all mobs as running in place.  Thrown/dropped items are also floating in the air.  The problem fixes itself for a minute or two each hour at random.  They are unable to see mobs in their correct serverside positions;  they are attacked by "invisible" entities.  Players are stuck in place for them as well.  On my screen one of them will be standing in place and I can attack them and deal damage on their end even when they are many blocks away (even 30+).

This has occurred with completely fresh installs of 1.20pre13 and now 1.20-RC1

I tried to play a fresh install of 1.19.8 on my other server (same host pc) and the problem was not present in that version for anyone.

### How to reproduce

I have been unable to figure out what causes the issue myself, after PC restarts, server restarts, and fresh client installs.  It simply affects some players and does not affect others.  All of us have tried switching between wired and wireless internet connections.

### Logs

[client-main.log](https://github.com/user-attachments/files/17867683/client-main.log)

**Comment by maltiez2** [comment]
Can be related to high ping

**Comment by iForged** [comment]
I can confirm that this issue does exist, but on my server, it's experienced by a couple of players who are connecting from another country and have an ADSL Internet connection. Since my server also uses version 1.20-pre13, I assume that it is more demanding on the connection quality between the client and the server. However, I cannot rule out that this could also be due to bugs related to the early version of the game.

**Comment by gerbilfat** [comment]
Confirming that this is also occurring on a friend's vanilla 1.20-pre13 server. There seems to be some kind of desync issue between the client and the server where mob and item locations are not correctly tracked.  This is leading to issues where items will float midair (i.e. when chopping down trees), and the player will take damage due to an 'invisible' mob that is displayed client-side as a farther distance away. 

Combat is impossible in this situation, and as a result, pre13 is not very playable on SMP. Areas with a lot of hostile mobs like caves or forests are death traps. Temporal storms have the same issue. 

I don't think this is related to ping as it occurs even when connecting via LAN and not over the internet.

**Comment by Denys10** [comment]
Confirming that this is also occurring on vanilla 1.20-pre13 server.

**Comment by Denys10** [comment]
Processador	Intel(R) Xeon(R) CPU E5-2690 v4 @ 2.60GHz   2.60 GHz
RAM instalada	63.8 GB
Edição	Windows 11 Pro
Versão	24H2

**Comment by LynnSnow13** [comment]
I also have two players having this same issue on our hosted server running 1.20pre13. There are three other players not having this issue. The two people with this same issue both live in the same house, but after walking them through checking their ping with packetstats, it doesn't seem to be the quality of their connection.

They did not have this issue with our 1.19 server.

**Comment by gunnerysurgeon** [comment]
Now that rc1 is out I am going to test if the problem still exists.  Will report back tonight or tomorrow.

**Comment by LynnSnow13** [comment]
Our crew still has the same members being affected by this bug on rc1.

**Comment by jsknnr** [comment]
I ran into this same issue last night on my dedicated 1.19.8 server running on actual server hardware in my house, so I don't think this is a latency issue. I restarted the server (container) and it was fine again.

**Comment by Denys10** [comment]
Now rc1 problem still exists  !!!

**Comment by Denys10** [comment]
On the 1.19 servers I can log in normally with or without mods, without problems, but on pre 13 and RC1 it is completely bugged with ping errors between player and server.

**Comment by Denys10** [comment]
I installed a new instance of Windows 10 and the problem persists, it's not the operating system, it's probably something related to the game and my hardware.

**Comment by gunnerysurgeon** [comment]
It is still occurring for me as well on RC1.

**Comment by RainyLupus** [comment]
Staying in one place for a period of time allows items to drop and mobs to move
Moving 10-20 blocks makes everything freeze again

**Comment by RainyLupus** [comment]
Possible Temp Fix. If you have a VPN it seems to have fixed the issue for me for the time being, it seems that some kind of port forwarding issue is to blame for this.

**Comment by riley00** [comment]
Can confirm still happening on 1.20-rc1. Tested with two different servers on different connections.

Doesn’t happen for anyone connected locally in my experience. I think hardware power might be irrelevant here- it happened on both my very weak server and a large server with excellent hardware.

**Comment by Khaosmatic** [comment]
Can also confirm its happening for me but not the two other people on my server. We all have 35ms or below connection and I'm not getting any ping spikes or anything.

Its a 3rd party server which according to the control panel is running smooth and I cant see any standout errors but there are some things, especially this which mentions entities:

![image](https://github.com/user-attachments/assets/9bb63344-88a2-4450-9976-3bec605ef141)

**Comment by Khaosmatic** [comment]
> On my server it seems i'm the only player on my server who is having this problem, I thought it was a tick rate issue at first and tried changing that but it didn't fix it. I tried reinstalls and the likes but nothing has changed. Animals far away some times works fine but the closer to the last known position the more likely the animals are frozen in place

This is exactly my case too. I run towards animals and then they freeze.

**Comment by JuleMcCormack** [comment]
We are also getting the same issues on our server

**Comment by LynnSnow13** [comment]
> Can also confirm its happening for me but not the two other people on my server. We all have 35ms or below connection and I'm not getting any ping spikes or anything.
> 
> Its a 3rd party server which according to the control panel is running smooth and I cant see any standout errors but there are some things, especially this which mentions entities:
> 
> ![image](https://private-user-images.githubusercontent.com/8474140/393359459-9bb63344-88a2-4450-9976-3bec605ef141.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzM1MTExMjIsIm5iZiI6MTczMzUxMDgyMiwicGF0aCI6Ii84NDc0MTQwLzM5MzM1OTQ1OS05YmI2MzM0NC04OGEyLTQ0NTAtOTk3Ni0zYmVjNjA1ZWYxNDEucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTIwNiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDEyMDZUMTg0NzAyWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MjJhMjdjNWRiZDUyZmM5ZjYyMjE4ZGFmMmY2MDgyODJjOTZiZDRlN2RlYWExOTMxZjZhY2I2NDEwY2Y2MjE1MiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.Tdl_AZZEaU50ypBpJz1ZKbRvfxprnI1XOkowWGsQP4U)

This is the same error the two people on my server pop when running up to the frozen mobs and interacting with them. The mobs will either do nothing or suddenly seem to teleport on their screen to where the mob is for everyone else.

Frustratingly, the mobs will appear frozen on their screen, but still be able to "chase" them and damage them. Invisible wolves, ha

**Comment by gunnerysurgeon** [comment]
> Can also confirm its happening for me but not the two other people on my server. We all have 35ms or below connection and I'm not getting any ping spikes or anything.
> 
> Its a 3rd party server which according to the control panel is running smooth and I cant see any standout errors but there are some things, especially this which mentions entities:
> 
> ![image](https://private-user-images.githubusercontent.com/8474140/393359459-9bb63344-88a2-4450-9976-3bec605ef141.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzM1NDc5NzIsIm5iZiI6MTczMzU0NzY3MiwicGF0aCI6Ii84NDc0MTQwLzM5MzM1OTQ1OS05YmI2MzM0NC04OGEyLTQ0NTAtOTk3Ni0zYmVjNjA1ZWYxNDEucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTIwNyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDEyMDdUMDUwMTEyWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NWY3ZTg2NWYxMjc4MzdlYmFkODEyMWE2MzJlMmIyZmY5NmI0YjZlOGQ1YjY5ZjExM2YyOWM0ODc1YzNlNDUzYSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.q5e9WmvEtxWodXLG28oCPkFpS6rjol2DheRiiaW_Z_Y)

This must be because the client is trying to interact with the entity on their screen but serverside the entity is nowhere near them.

**Comment by jaksonkallio** [comment]
Same issues on our server! I posted this on the forum, but seems like the GitHub issues are more active.

Since updating to 1.20.0-rc.1, players are able to destroy blocks, but not pick up items. Also, players appear to be stuck in place when they actually are moving around. Server was working flawlessly in 1.19.8.

Video of the issue: https://youtu.be/bpe1aRe5dRA

As required, I had to update our infra to support the new UDP port which is the only infra change that I am aware of. The server is hosted in a Docker container behind Traefik reverse proxy. I tried restarting the server, rebuilding the server Docker image, creating fresh saves, and disabling client firewalls.

In the server logs I noticed that there were a few lines saying "server overloaded", maybe the new server-side physics increased the required resources? However while running the server, CPU never went above 20% usage.

**Comment by AreYSerious** [comment]
bump

**Comment by Darkx352** [comment]
We are having this issue for some players on our MP server. 
doesn't matter there location from the server as some are on the same coastline and they will have the issue and other will be across the pond and have zero issues. 
I am wondering if it has something to do with the game using a UPD connection for the players instead of a TCP connection in the port forwarding system.

**Comment by RainyLupus** [comment]
Looking into UPD it could be that the server isn't getting the packets needed to have a stable connection. Leading to these frozen animals or items. There is also not a way to see Packet loss on UPD from what i been reading. Could also be that the server is getting overloaded with packets.

**Comment by Denys10** [comment]
> Olhando para o UPD, pode ser que o servidor não esteja recebendo os pacotes necessários para ter uma conexão estável. Levando a esses animais ou itens congelados. Também não há uma maneira de ver a perda de pacotes no UPD pelo que tenho lido. Também pode ser que o servidor esteja ficando sobrecarregado com pacotes.

It does not occur with the entire server, this occurrence is only with some specific players, this is certainly something related to connectivity, however the internet works normally on 1.19 servers

**Comment by tyronx** [comment]
in 1.20 entity movement is send via UDP, so yea there is a connectivity problem with UDP

**Comment by Wesmania** [comment]
Confirming the same issue. Some stuff I found blindly poking around with wireshark:
* I have a linux server and linux client exhibiting the issue.
* Client-side, I send UDP packets regularly, but I only ever receive 4 UDP packets from the server, no more ever come.
* Server-side, I receive the packets from the client and send some response packets. These show in tshark like this:
  ```
  7 0.068106206 <server ip> → <NATed client IP> IPv4 1514 Fragmented IP protocol (proto=UDP 17, off=0, ID=b791)
  ```
* Investing these response packets, it looks like every single UDP packet sent from the server has a "more packets" flag set. I get the feeling this is wrong. Could it be some issue in some .NET networking library?

**Comment by Torc1998** [comment]
I have the same problem...
While trying to fix a crash issue, I noticed something.
I don't know if this could be related.

When I connect to the server with the view distance set to 32, the server only generates 1 chunk and the animals walk around normally. Could the server generate something that spams the UDP port outside of the chunk?

**Comment by Wesmania** [comment]
Would it make sense to reduce max data size sent at once by the game via UDP to ~1200 or ~1480 bytes so that it always fits in an MTU? I guess fragmented packets are just getting filtered by firewalls.

**Comment by Th3Dilli** [comment]
hmm that all sounds it could be all related to large UDP packages getting split and those are then not delivered correctly.

On discord someone reported they only run into the issue with a certain amount of entities.

 I can add some code to keep the packages smaller and then we can test if that helps, ty so far

**Comment by Orangutanion** [comment]
Here is video evidence. Ping was about 144ms which should definitely be a supported ping value: https://imgur.com/a/phcR4tf

**Comment by Wesmania** [comment]
I tested rc.3 for around 5 minutes, I can see an improvement. Animals and physics didn't lag at all.

**Comment by Khaosmatic** [comment]
So far so good on RC.4. I can actually fight and hunt now! :)

**Comment by Wesmania** [comment]
I played for a good few hours on rc.4 now with absolutely no entity lag.

**Comment by radfast** [comment]
We believe these issues with entity physics packets are fully fixed by [game version 1.20.0-rc.6](https://www.vintagestory.at/blog.html/news/v120-rc6-lore-update-the-journey-r392/) available since yesterday, so I think this can now be closed

(One new issue apparently rises, #4796, but that is hopefully not common, only seen once so far on a server restart.)
